krabs/krabs/ut.hpp

```diff
@@ -152,14 +160,22 @@ namespace krabs { namespace details {
 
                 //allocate + size of expected events in filter
                 DWORD size = FIELD_OFFSET(EVENT_FILTER_EVENT_ID, Events[provider.second.provider_filter_event_ids_.size()]);
-                filterMemoryPtr = std::make_unique<BYTE[]>(size);
+                std::vector<BYTE> filterMemoryPtr(size);
 
-                auto filterEventIds = reinterpret_cast<PEVENT_FILTER_EVENT_ID>(filterMemoryPtr.get());
+                auto filterEventIds = reinterpret_cast<PEVENT_FILTER_EVENT_ID>(&(filterMemoryPtr[0]));
                 filterEventIds->FilterIn = TRUE;
                 filterEventIds->Count = static_cast<USHORT>(provider.second.provider_filter_event_ids_.size());
-                for (int index = 0; index < filterEventIds->Count; ++index) {
-                    filterEventIds->Events[index] = provider.second.provider_filter_event_ids_[index];
+
+                auto index = 0;
+                std::set<unsigned short>::iterator provider_filter_iterator;
+                for (provider_filter_iterator = provider.second.provider_filter_event_ids_.begin();
```

Comment:
An alternative way to do this would be to use the [range-based for-loop](https://en.cppreference.com/w/cpp/language/range-for) that is available to iterators:

```suggestion
                for (auto filter : provider.second.provider_filter_event_ids_) {
                     filterEventIds->Events[index] = filter;
                     index++;
                }
```

**Note: I didn't test that code, that's just a rough example.**